### PR TITLE
Pool proxy storage

### PR DESCRIPTION
**Pull request by CodeSandwich**
Part of #16. Adds capability to store proxy weights in ReceiverWeights. It'll start being used in the next PR.

### CODE REVIEW

File: contracts/PoolTest.sol [hunk 1]
```diff
@@ -12,56 +12,81 @@ contract ReceiverWeightsTest {
...
         uint256 iterationGasUsed = 0;
         while (true) {
             // Each step of the non-pruning iteration should yield the same items
             address oldReceiver = receiver;
-            uint32 weight;
+            uint32 weightReceiver;
+            uint32 weightProxy;
             uint256 gasLeftBefore = gasleft();
-            (receiver, weight) = receiverWeights.nextWeightPruning(oldReceiver);
+            (receiver, weightReceiver, weightProxy) = receiverWeights.nextWeightPruning(receiver);
```
Comment (geigerzaehler) [inline contracts/PoolTest.sol:61]: This is a bit clearer to me
```suggestion
            (receiver, weightReceiver, weightProxy) = receiverWeights.nextWeightPruning(oldReceiver);
```

File: test/pool.test.ts [hunk 2]
```diff
@@ -744,107 +770,154 @@ describe("ReceiverWeights", function () {
...
+    assert((await weights_test.weightReceiverSumDelta()).eq(0));
+    assert((await weights_test.weightProxySumDelta()).eq(1));
+    const weights = await weights_test.getReceiverWeightsIterated();
+    assert(weights.length == 1);
+    assert(weights[0].receiver == addr1);
+    assert(weights[0].weightReceiver == 0);
+    assert(weights[0].weightProxy == 1);
+  });
+
+  it("Allows removing items with only proxy weights set", async function () {
```
Comment (geigerzaehler) [inline test/pool.test.ts:905]: Given the increased size of the ReceiverWeights tests it might make sense to move them into their own test file.

Comment (CodeSandwich) [reply test/pool.test.ts]: Great idea, fixed

File: contracts/PoolTest.sol (hunk 1 above)
Comment (CodeSandwich) [reply contracts/PoolTest.sol]: Fixed

File: contracts/PoolTest.sol [hunk 3]
```diff
@@ -12,56 +12,81 @@ contract ReceiverWeightsTest {
 
     ReceiverWeights private receiverWeights;
     ReceiverWeightIterated[] private receiverWeightsIterated;
-    int256 public receiverWeightsSumDelta;
+    int256 public weightReceiverSumDelta;
+    int256 public weightProxySumDelta;
```
[the remark this answers is not in the document]
Comment (CodeSandwich) [reply contracts/PoolTest.sol]: Fixed

File: contracts/Pool.sol [hunk 4]
```diff
@@ -405,31 +407,41 @@ struct ReceiverWeights {
...
     }
 
     address internal constant ADDR_ROOT = address(0);
     address internal constant ADDR_UNINITIALIZED = address(0);
     address internal constant ADDR_END = address(1);
 
-    /// @notice Return the next non-zero receiver weight and its address.
-    /// Prunes all the zeroed items found between the current and the next receivers.
+    /// @notice Return the next non-zero receiver or proxy weight and its address.
+    /// Prunes all the fully zeroed items found between the current and the next receivers.
```
[the remark this answers is not in the document]
Comment (CodeSandwich) [reply contracts/Pool.sol]: Fixed

**Review by geigerzaehler — APPROVED** [review]
:sun_with_face:

**Pull request merged**